### PR TITLE
Bump sabre dependencies

### PR DESCRIPTION
**Pull request by phil-davis**
## Description
1) A release of `sabre/http`
```
$ composer update
Loading composer repositories with package information
Updating dependencies
Lock file operations: 0 installs, 2 updates, 0 removals
  - Upgrading nikic/php-parser (v4.13.0 => v4.13.1)
  - Upgrading sabre/http (5.1.1 => 5.1.2)
Writing lock file
```

https://github.com/sabre-io/http/releases/tag/5.1.2

2) Then I made releases of some other sabre dependencies
```
$ composer update
Loading composer repositories with package information
Updating dependencies
Lock file operations: 0 installs, 3 updates, 0 removals
  - Upgrading sabre/event (5.1.2 => 5.1.3)
  - Upgrading sabre/vobject (4.3.5 => 4.3.6)
  - Upgrading sabre/xml (2.2.3 => 2.2.4)
Writing lock file
```

3) I missed bumping a VERSION constant in the releases, so I had to make another patch release of each:
```
$ composer update
Loading composer repositories with package information
Updating dependencies
Lock file operations: 0 installs, 4 updates, 0 removals
  - Upgrading sabre/event (5.1.3 => 5.1.4)
  - Upgrading sabre/http (5.1.2 => 5.1.3)
  - Upgrading sabre/vobject (4.3.6 => 4.3.7)
  - Upgrading sabre/xml (2.2.4 => 2.2.5)
Writing lock file
```

Note: `nikic/php-parser` is a dev tool, so no changelog needed for that.

## How Has This Been Tested?
CI

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:
- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [x] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

File: changelog/unreleased/PHPdependencies20210721onward [hunk 1]
```diff
@@ -16,6 +16,10 @@ The following have been updated:
 - phpseclib/phpseclib (3.0.9 to 3.0.11)
 - pimple/pimple (3.2.3 to 3.5.0)
 - punic/punic (3.6.0 to 3.6.1)
+- sabre/event (5.1.2 t0 5.1.4)
```
Comment (JammingBen) [inline changelog/unreleased/PHPdependencies20210721onward:19]:
```suggestion
- sabre/event (5.1.2 to 5.1.4)
```

File: changelog/unreleased/PHPdependencies20210721onward [hunk 2]
```diff
@@ -16,6 +16,10 @@ The following have been updated:
...
+- sabre/http (5.1.1 to 5.1.3)
+- sabre/vobject (4.3.5 t0 4.3.7)
```
Comment (JammingBen) [inline changelog/unreleased/PHPdependencies20210721onward:21]:
```suggestion
- sabre/vobject (4.3.5 to 4.3.7)
```

**Review by JammingBen — CHANGES_REQUESTED** [review]
Minor typo, LGTM beside that.

File: changelog/unreleased/PHPdependencies20210721onward (hunk 1 above)
Comment (phil-davis) [reply changelog/unreleased/PHPdependencies20210721onward]: fixed

File: changelog/unreleased/PHPdependencies20210721onward (hunk 2 above)
Comment (phil-davis) [reply changelog/unreleased/PHPdependencies20210721onward]: fixed

**Review by JammingBen — APPROVED** [review]

**Pull request merged**